tests/integration_tests/functional/test_cpu_features_aarch64.py

```diff
@@ -3,83 +3,63 @@
 
 import os
-import platform
-import re
 
 import pytest
 
-import framework.utils_cpuid as cpuid_utils
 from framework import utils
 from framework.properties import global_props
 from framework.utils_cpuid import CPU_FEATURES_CMD, CpuModel
 
-PLATFORM = platform.machine()
+pytestmark = pytest.mark.skipif(
+    global_props.cpu_architecture != "aarch64", reason="Only run in aarch64"
+)
 
-DEFAULT_G2_FEATURES = set(
+G2_FEATS = set(
     (
         "fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp "
         "asimdhp cpuid asimdrdm lrcpc dcpop asimddp ssbs"
-    ).split(" ")
+    ).split()
 )
 
-DEFAULT_G3_FEATURES_5_10 = DEFAULT_G2_FEATURES | set(
-    "sha512 asimdfhm dit uscat ilrcpc flagm jscvt fcma sha3 sm3 sm4 rng dcpodp i8mm bf16 dgh".split(
-        " "
-    )
+G3_FEATS = G2_FEATS | set(
+    "sha512 asimdfhm dit uscat ilrcpc flagm jscvt fcma sha3 sm3 sm4 rng dcpodp i8mm bf16 dgh".split()
 )
 
-DEFAULT_G3_FEATURES_WITH_SVE_AND_PAC_5_10 = DEFAULT_G3_FEATURES_5_10 | set(
-    "paca pacg sve svebf16 svei8mm".split(" ")
-)
+G3_SVE_AND_PAC = set("paca pacg sve svebf16 svei8mm".split())
 
-DEFAULT_G3_FEATURES_V1N1 = DEFAULT_G2_FEATURES
 
+def test_guest_cpu_features(uvm_any):
+    """Check the CPU features for a microvm with different CPU templates"""
 
-def _check_cpu_features_arm(test_microvm, guest_kv, template_name=None):
-    expected_cpu_features = {"Flags": []}
-    match cpuid_utils.get_cpu_model_name(), guest_kv, template_name:
-        case CpuModel.ARM_NEOVERSE_N1, _, "v1n1":
-            expected_cpu_features = DEFAULT_G2_FEATURES
-        case CpuModel.ARM_NEOVERSE_N1, _, None:
-            expected_cpu_features = DEFAULT_G2_FEATURES
+    vm = uvm_any
+    expected_cpu_features = set()
+    match global_props.cpu_model, vm.cpu_template_name:
+        case CpuModel.ARM_NEOVERSE_N1, "v1n1":
+            expected_cpu_features = G2_FEATS
+        case CpuModel.ARM_NEOVERSE_N1, None:
+            expected_cpu_features = G2_FEATS
 
         # [cm]7g with guest kernel 5.10 and later
-        case CpuModel.ARM_NEOVERSE_V1, _, "v1n1":
-            expected_cpu_features = DEFAULT_G3_FEATURES_V1N1
-        case CpuModel.ARM_NEOVERSE_V1, _, "aarch64_with_sve_and_pac":
-            expected_cpu_features = DEFAULT_G3_FEATURES_WITH_SVE_AND_PAC_5_10
-        case CpuModel.ARM_NEOVERSE_V1, _, None:
-            expected_cpu_features = DEFAULT_G3_FEATURES_5_10
-
-    _, stdout, _ = test_microvm.ssh.check_output(CPU_FEATURES_CMD)
-    flags = set(stdout.strip().split(" "))
-    assert flags == expected_cpu_features
+        case CpuModel.ARM_NEOVERSE_V1, "v1n1":
+            expected_cpu_features = G2_FEATS
+        case CpuModel.ARM_NEOVERSE_V1, "aarch64_with_sve_and_pac":
+            expected_cpu_features = G3_FEATS | G3_SVE_AND_PAC
+        case CpuModel.ARM_NEOVERSE_V1, None:
+            expected_cpu_features = G3_FEATS
 
+    guest_feats = set(vm.ssh.check_output(CPU_FEATURES_CMD).stdout.split())
```

Comment:
nit: should we `strip()` the `stdout` as well to avoid trailing blank characters resulting in empty strings? or does `split()` already remove those?

Reply:
Yeah, `.split()` behaves different than `.split(" ")` and does what you want commonly:

```
"     a   b    c    ".split()
['a', 'b', 'c']
```